app/services/users.js

```diff
@@ -0,0 +1,24 @@
+const { inspect } = require('util');
+const bcrypt = require('bcrypt');
+
+const errors = require('../errors');
+const Users = require('../models/users');
+const logger = require('../logger');
+
+exports.createUser = newUser =>
+  Users.findOrCreate(newUser)
+    .then(user => {
+      const created = user[1];
+      if (created) return user[0];
+      throw errors.emailExistingError('Error, email already exists');
```

Comment:
Pay attention to this, because if you throw this error (emailExistingError), the error will catch for the next catch who throws another error (userSignupError).